src/Microsoft.Dnx.Runtime/NuGet/SemanticVersion.cs

```diff
@@ -319,7 +272,11 @@ public int CompareTo(SemanticVersion other)
 
         public override string ToString()
         {
-            return OriginalString;
+            var revision = Version.Revision > 0 ? ("." + Version.Revision.ToString(CultureInfo.InvariantCulture)) : string.Empty;
+            var specialVer = !string.IsNullOrEmpty(SpecialVersion) ? ("-" + SpecialVersion) : string.Empty;
+
+            // SemanticVersion normalizes the missing components to 0.
+            return $"{Version.Major}.{Version.Minor}.{Version.Build}{revision}{specialVer}";
```

Comment:
1. Calling `ToString()` on one version object multiple times is common. We shouldn't repeat the work each time when `ToString()` is called.
2. Also note that sometimes we create version objects just to do some comparison and we never call ToString() on them. We should avoid allocating strings in this case.

You may want to use lazy initialization here.